src/main/java/NetworkView.java

```diff
@@ -63,7 +63,9 @@ void paintConnections(Graphics g) {
         g.setColor(Color.BLUE);
         for (GraphNode node : listNodes) {
             for (GraphNode connection : node.getConnections()) {
-                if (drawnConnections.get(node) != connection) {
+                if (drawnConnections.containsKey(node)) {
+                    if (drawnConnections.get(node).equals(connection)) ;
```

Comment:
what do these if statements do?

Reply:
The inner if statement checks if the connection has been drawn, if so, it doesn't draw it and moves on to the next. The outer if statement is necessary because without it, the inner if statement throws null pointer exceptions when the get() method is called for a node which isn't in the hashmap.

Reply:
why not do 
```
if (!drawnConnections.containsKey(node)) {
                    g.drawLine(node.getX(), node.getY(), connection.getX(), connection.getY());
                    drawnConnections.put(node, connection);
                }
```

Reply:
Because drawnConnections is a hashmap in which the keys and values are pairs of nodes whose connections have been drawn. Let's say node 1 is connected to nodes 2 and 3, and the connection between 1 and 2 has been drawn, but not 1 and 3. Only checking if node 1 is a key will not show that 1 and 3's connection has not yet been drawn. But now that you made me think about this, I'm realizing that it's still not right. I think it needs to be this: if (drawnConnections.get(node).equals(connection) || drawnConnections.get(connection).equals(node));
Because when the node is 2 and the connection is 1, we don't want to redraw the connection between 1 and 2.